app/models/setting.rb

```diff
@@ -270,6 +270,11 @@ def self.load_defaults
   end
 
   def self.set(name, description, default, full_name = nil, value = nil)
+    if block_given?
+      self.class_eval do
```

Comment:
creates a  dynamic instance method, which it's lighter solution then save it on DB, plus it allow the collection be dynamic.